src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -185,25 +208,44 @@ public static function get_core_data() {
 	 *
 	 * @param string $origin Data source for which to cache the blocks.
 	 *                       Valid values are 'core', 'blocks', 'theme', and 'user'.
+	 * @param bool   $persistant_cache Optional. Whether to check the persistent cache. Default false.
 	 * @return bool True on success, false otherwise.
 	 */
-	protected static function has_same_registered_blocks( $origin ) {
+	protected static function has_same_registered_blocks( $origin, $persistant_cache = false ) {
```

Comment:
@joemcgill It was a bit of a design decision. I wanted theme data to be checked against the persistent list as we assume the blocks will not change before the expiry, whereas blocks from `user` can change dynamically at any instance. 
`$blocks_cache` is just a static cache, and it always contains blank arrays of origins in the first call.

If we use the original `$blocks_cache` the first time `has_same_registered_blocks` is called, it will always return `false` as initially `$blocks_cache` is set to blank arrays, and `has_same_registered_blocks` sees all blocks as new blocks.

This version of `has_same_registered_blocks` can't be used with persistent cache inside `get_theme_data` as it will assume the block has changed and will update static::theme, thereby also calling `set_site_transient()`.  

Now I can't update `$blocks_cache` with  precipitant value even when  `$can_use_cached` is true as it will update 
`$blocks_cache` with persistent value for other origins, too, like `user` or `block`, which depended on change based on `has_same_registered_blocks` or not.

Let me check again to see if there is a scope for using just one variable.